### PR TITLE
Update Client.js in order to detect login in new WhatsAppWeb

### PR DESCRIPTION
**Pull request by elhumbertoz**
New WhatsAppWeb selector for login sucess

# PR Details

## Description

Detect new WhatsApp Web Login event

## Related Issue

No Login detect

## Motivation and Context

Client don't detect login event

## How Has This Been Tested

Yes I tested

## Types of changes

I change INTRO_IMG_SELECTOR value in Client.js

### CODE REVIEW

**Review by Robonator100 — APPROVED** [review]

**Review by rayss868 — APPROVED** [review]

**Review by caturarp — APPROVED** [review]
fitted the update

**Review by rizkydoang — APPROVED** [review]

**Review by fabiomello — APPROVED** [review]

**Review by d3v7 — APPROVED** [review]

**Review by zKriguer — APPROVED** [review]
Good update, old logo is broken, UP

**Review by Kyori-kyo — APPROVED** [review]
actual icon is broken
this pr fixes it

**Comment by jrocha** [comment]
works!

**Comment by leoneDuarte** [comment]
Funcionou para min essa correção!

**Review by FacundoSuarez-46 — APPROVED** [review]
Works perfect!

**Review by vzolin — APPROVED** [review]
Tested. After change the 'ready' event was emmited

**Review by trktml — APPROVED** [review]

**Review by matifanger — APPROVED** [review]
lgtm

**Comment by viniciuspereiras** [comment]
Works like a charm! tks

**Review by sadewoimam — APPROVED** [review]

**Review by themazim — APPROVED** [review]

**Review by TemaSM — APPROVED** [review]
Thanks for fix, it works

**Comment by musrais** [comment]
Yes, it's work!

**Review by wszf — APPROVED** [review]

**Comment by ricardolucena** [comment]
It is not recovering from the previous session?

**Comment by iamvinny** [comment]
It works, please approve the PR.

**Comment by fpiekarski** [comment]
It works, please approve the PR.

File: src/Client.js
Comment (eduarantes) [inline src/Client.js:1]: it's work!

**Review by eduarantes — APPROVED** [review]

**Review by TuiNascimento — APPROVED** [review]

**Review by halimbrian — APPROVED** [review]
LGTM

**Comment by jaenudin86** [comment]
tanks works, awesome

**Comment by MrPolloo** [comment]
awesome! it works with me too

**Comment by bmmangukiya** [comment]
This Works

**Review by iglesiasfernando — COMMENTED** [review]
This workaround its working

**Comment by ndeitch** [comment]
I did this workaround

```js
const waitPupPage = () => {
  if (client.pupPage) {
    client.pupPage.waitForSelector("div[role='textbox']").then(() => {
      client.pupPage.evaluate(() => (document.body.innerHTML = '<div data-icon="chat" />'))
    })
  } else {
    setTimeout(waitPupPage, 1000)
  }
}

waitPupPage()
```

It creates expected selector once it detects the new one.

**Review by AlloryDante — APPROVED** [review]
Approve

**Comment by almacres10** [comment]
this solved my issue three days ago, but today the issues shows again. any other solutions?

**Comment by zvictor** [comment]
> I did this workaround
> 
> ```js
> const waitPupPage = () => {
>   if (client.pupPage) {
>     client.pupPage.waitForSelector("div[role='textbox']").then(() => {
>       client.pupPage.evaluate(() => {
>         const elem = document.createElement('div')
>         elem.setAttribute('data-icon', 'chat')
>         document.body.appendChild(elem)
>      })
>     })
>   } else {
>     setTimeout(waitPupPage, 1000)
>   }
> }
> 
> waitPupPage()
> ```
> 
> It creates expected selector once it detects the new one.

@ndeitch I like your way of thinking. This is ugly and pretty at the same time. 😅
Pretty because I don't need to patch the lib, ugly because it adds lines that shouldn't be in my codebase.
I am gonna use it.

**Review by viniciuspereiras — APPROVED** [review]
Worked

**Review by Marqitos — APPROVED** [review]
This change make that login works again

**Review by Marqitos — APPROVED** [review]
This change make that login works again

**Review by wiz64 — APPROVED** [review]
Should be merged, fixed my issue

File: src/Client.js [hunk 1]
```diff
@@ -172,7 +172,7 @@ class Client extends EventEmitter {
             }
         );
 
-        const INTRO_IMG_SELECTOR = '[data-icon=\'chat\']';
```
Comment (tuyuribr) [inline src/Client.js:175]: Should be changed to `search` right? @elhumbertoz (Better query)

Comment (tuyuribr) [reply src/Client.js:175]: image

**Review by tuyuribr — APPROVED** [review]

**Review by shirser121 — APPROVED** [review]
Thanks

**Pull request merged**